app/src/main/java/de/danoeh/antennapod/activity/ImportExportActivity.java

```diff
@@ -123,6 +128,24 @@ private void restoreFrom(Uri inputUri) {
         }
     }
 
+    private static final byte[] SQLITE3_MAGIC = "SQLite format 3\0".getBytes();
+    private boolean validateDB(Uri inputUri) throws IOException {
+        boolean ok = false;
+        try (InputStream inputStream = getContentResolver().openInputStream(inputUri)) {
+            byte[] magicBuf = new byte[16];
+            if (inputStream.read(magicBuf) == magicBuf.length) {
+                ok = Arrays.equals(SQLITE3_MAGIC, magicBuf);
+            }
+        }
+
+        if (!ok) {
+            Log.i(TAG, "Not a SQLite3 file: " + inputUri.toString());
+            Snackbar.make(findViewById(R.id.import_export_layout), R.string.import_bad_file, Snackbar.LENGTH_SHORT).show();
```

Comment:
I'd prefer an alert instead of a toast. The later can be easily missed.
Great job!